### ISSUE TITLE
Loopy Belief inference is only implemented for a single node.

### ISSUE THREAD

**Issue by tawheeler** [sisl/BayesNets.jl#48]
I introduced a new inference API:

```
abstract InferenceMethod

"""
Infer p(query|evidence)
 - inference on a DiscreteBayesNet will always return a DataFrame factor over the evidence variables
"""
infer(im::InferenceMethod, bn::BayesNet, query::Vector{NodeName}; evidence::Assignment=Assignment())
```

Likelihood and exact inference work perfectly under this.
Loopy belief, however, only takes a `query::NodeName`, when a `Vector{NodeName}` would be ideal.

**Comment by tawheeler** [comment]
@hamzaelsaawy - if you could address this that'd be excellent.

**Comment by hamzaelsaawy** [comment]
With regards to this and #49, I have a new Factor type, and an inference type.
[I'm still testing/integrating though](https://github.com/hamzaelsaawy/BayesNets.jl/tree/inference)

I've been struggling with this; I don't think LBP supports multiple queries, since it just returns the marginal of any one node (given the evidence) by message passing, not the joint of multiple nodes. I think ... 

I could be misunderstanding basic probability though

**Comment by mykelk** [comment]
If you have multiple query variables, you can turn them into a vector variable, and then proceed with the LBP algorithm.

**Comment by henripal** [comment]
agreed that LBP won't return the joint but I think that one would expect a LBP call with multiple query nodes to return the marginals for each node, which you already have after the LBP loop terminates?

**Comment by tawheeler** [comment]
We could just take the factor product over all resulting marginal factors that correspond to query variables.
